### PR TITLE
enables to configure an interval to check for updates

### PR DESCRIPTION
**Pull request by AngeloBonini**

### CODE REVIEW

**Review by rodbecc — DISMISSED** [review]

File: scripts/self_update.ts [hunk 1]
```diff
@@ -1,15 +1,44 @@
...
+        lastCheckUpdateFileName,
+      ),
+    );
+
+    const daysSinceLastCheckUpdate = difference(lastCheckUpdateDate, new Date(today), {
+      units: ['days'],
+    }).days;
+
+    daysSinceLastCheckUpdate &&
+      daysSinceLastCheckUpdate >= +checkUpdateIntervalEnv && checkNewVersion();
```
Comment (vvasc) [inline scripts/self_update.ts:34]: where are we using this? It seems to be only conditional without having a variable to hold its value. Shouldn't be better to have a variable and split this into multiple variables to be more semantic?

__
[sema-logo](https://semasoftware.com/gh) &nbsp;**Summary:** :hammer_and_wrench: This code needs a fix&nbsp; | &nbsp;**Tags:** Inefficient

File: scripts/self_update.ts [hunk 2]
```diff
@@ -54,3 +83,13 @@ async function update() {
 
   await (changedToMain ? runPullOriginMainCmd.status().then(() => printLogo()) : Promise.reject(textDecoder(error)));
 }
+
+async function storeLastCheckUpdateDate() {
+  try {
```
Comment (vvasc) [inline scripts/self_update.ts:88]: As we've got the try-catch inside the `shouldCheckForUpdates` function, we might don't need this one. What do you think? Is there any flow that can trigger the catch here?

__
[sema-logo](https://semasoftware.com/gh) &nbsp;**Summary:** :hammer_and_wrench: This code needs a fix

**Review by vvasc — CHANGES_REQUESTED** [review]
Other than that, this is a very small function so I think we can have good coverage of unit tests here.

And, shouldn't be nice to have proper documentation on how to set the date in the las_check_update.txt file?

**Comment by vvasc** [comment]
besides, nice first PR @AngeloBonini !!!

File: scripts/self_update.ts (hunk 1 above)
Comment (vvasc) [reply scripts/self_update.ts:34]: I think you can put an if there

Comment (AngeloBonini) [reply scripts/self_update.ts:34]: ok, I will work on the changes

**Comment by AngeloBonini** [comment]
I spoke to @rodbecc and because I will change the implementation of `checkUpdateIntervalEnv` to a .env file, the tests and the proper documentation will be added on another issue. I made the changes on the conditional.

**Comment by AngeloBonini** [comment]
I made the changes on the conditional statement

**Review by rodbecc — APPROVED** [review]

**Pull request merged**